script.js

```diff
@@ -1,4 +1,185 @@
+var gameMode_ROLL_DICE = "GAME_ROLL_DICE";
+var gameMode_CHOOSE_DICE = "CHOOSE_DICE";
+var gameMode_COMPARE_SCORES = "COMPARE_SCORES";
+var gameMode = gameMode_ROLL_DICE;
+var diceRollResults = [];
+var allPlayerResults = [];
+var playerCount = 1;
+var output = "";
+
+///////////////////////////////////
+// 1. DICE ROLL FUNCTION
+///////////////////////////////////
+var rollDice = function () {
+  // produce a decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+  // remove the decimal
+  var randomInteger = Math.floor(randomDecimal);
+  // add 1 to get a number between 1 and 6 inclusive
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+///////////////////////////////////
+// 2. ROLLS DICE TWICE
+///////////////////////////////////
+var playerRollDice = function () {
+  for (counter = 0; counter < 2; counter += 1) {
+    diceRollResults.push(rollDice());
+  }
+  return (
+    "Welcome Player " +
+    playerCount +
+    ", you rolled " +
+    diceRollResults[0] +
+    " for Dice one and " +
+    diceRollResults[1] +
+    " for Dice two. <br>Choose the order of the dice by inputting either '1' or '2'"
+  );
+};
+///////////////////////////////////
+// 3. CHOOSE ORDER OF DICE
+///////////////////////////////////
+var playerChooseDice = function (inputDice) {
+  var twodigit;
+  //if order is neither 1 nor 2 - input validation
+  if (inputDice != 1 && inputDice != 2) {
+    console.log("Error, input is not 1 or 2");
+    chooseDiceMessage =
+      "Please key in '1' or '2' only to choose your dice order. <br>Your dice one is: " +
+      diceRollResults[0] +
+      " and your dice two is: " +
+      diceRollResults[1];
+    return chooseDiceMessage;
+  }
+  //if player is player 1
+  if (playerCount == 1) {
+    //if order of dice is 1
+    if (inputDice == 1) {
+      twodigit = "" + Number(diceRollResults[0]) + Number(diceRollResults[1]);
```

Comment:
Nice way of concatenating the numbers, but when using methods like this, make sure you truly understand what is happening and your intention behind it! otherwise I would propose a different way maybe? :
```
Number(diceRollResults[0]) * 10 + Number(diceRollResults[1])
```